### PR TITLE
[KV Cache Injection] MPT

### PR DESCRIPTION
**Pull request by dbogunowicz**

### CODE REVIEW

File: src/sparseml/exporters/transforms/kv_cache/configs.py [hunk 1]
```diff
@@ -102,20 +105,24 @@ class Config:
...
     transpose_value_input=None,
     transpose_key_input=(0, 1, 3, 2),
-    multiply_batch_by_num_att_heads=True,
+    multiply_batch_by_num_att_heads=False,
 )
 
 
 def get_kv_cache_config(
     model_path: str,
-    supported_configs: List[BaseModel] = [OPT_CONFIG, CODEGEN_CONFIG, BLOOM_CONFIG],
```
Comment (dbogunowicz) [inline src/sparseml/exporters/transforms/kv_cache/configs.py:118]: We need to remove BLOOM, we are not supporting it officially in the nightly.

Comment (bfineran) [reply src/sparseml/exporters/transforms/kv_cache/configs.py:118]: if we have some internal testing, hidden support seems ok

File: src/sparseml/exporters/transforms/kv_cache/transforms_base.py [hunk 2]
```diff
@@ -97,6 +98,76 @@ def find_nodes_by_pattern(
...
+            input. The ordering of nodes is important. The first node in the list
+            must take the causal mask as input. The last node in the list will be
+            connected to the `causal_mask_input_child`.
+        :param initializers_to_add: an optional list of initializers to add to the model
+        :return: model with the causal mask adjusted
+        """
+
+        graph = ONNXGraph(model)
+
+        if nodes_to_add[0].input[0] != self.CAUSAL_MASK_NAME:
```
Comment (bfineran) [inline src/sparseml/exporters/transforms/kv_cache/transforms_base.py:144]: lgtm overall - let's discuss tomorrow if there's a way we can clean up this flow a bit. needing to format this list arg in a certain way seems a bit like hidden knowledge here

**Review by mgoin — COMMENTED** [review]
FYI I've tested this with MPT instruct, chat, and replit code to much success

**Review by bfineran — APPROVED** [review]

**Review by mgoin — APPROVED** [review]

**Pull request merged**